sklearn/metrics/pairwise.py

```diff
@@ -148,13 +153,17 @@ def check_pairwise_arrays(
     Returns
     -------
     safe_X : {array-like, sparse matrix} of shape (n_samples_X, n_features)
-        An array equal to X, guaranteed to be a numpy array.
+        An array equal to X, guaranteed to be an array.
 
     safe_Y : {array-like, sparse matrix} of shape (n_samples_Y, n_features)
-        An array equal to Y if Y was not None, guaranteed to be a numpy array.
+        An array equal to Y if Y was not None, guaranteed to be an array.
         If Y was None, safe_Y will be a pointer to X.
```

Comment:
As stated elsewhere, I would rather not change the public docstrings while Array API dispatch is still experimental and not enabled by default in scikit-learn.

The day we want to update the docstring to reflect Array API support, we should probably do so consistently across all public functions and classes, but first we would need to agree on how to update the documentation on docstring writing conventions.